### ISSUE TITLE
--warn-redundant-cast doesn’t warn for simple generic types

### ISSUE THREAD

**Issue by flying-sheep** [python/mypy#21796]
**Bug Report**
See reproducer

**To Reproduce**

https://mypy-play.net/?mypy=latest&python=3.12&flags=warn-redundant-casts&gist=f923e1fafbc60a2b7a2fa3c9c3ceb785

```python
from typing import TypeVar, cast

T = TypeVar("T")


def identity(xs: list[T]) -> list[T]:
    return xs


def f(xs: list[int]) -> list[int]:
    reveal_type(identity(xs))
    return cast(list[int], identity(xs))  # --warn-redundant-casts should trigger here

```

**Expected Behavior**

warns that the cast is redundant

**Actual Behavior**

doesn’t warn

**Your Environment**

- Mypy version used: 2.3.0
- Mypy command-line flags: `--warn-redundant-casts`
- Mypy configuration options from `mypy.ini` (and other config files): `warn-redundant-casts = true`
- Python version used: any

**Comment by Yoruxyv** [comment]
Hi! I’d be happy to take a look at this if no one is already working on it. I’ll first reproduce the issue on current main, then investigate the redundant-cast check and add a focused regression test with the fix. I’ll keep the change scoped to this behavior.
